### PR TITLE
Spawn multiple OpenStack servers at a time

### PR DESCRIPTION
**Pull request by sb10**
If scaling up to hundreds of servers, sequential server spawning could take many hours. Now it can take minutes.

### CODE REVIEW

File: jobqueue/scheduler/openstack.go [hunk 1]
```diff
@@ -591,41 +590,88 @@ func (s *opst) canCount(cmd string, req *Requirements, call string) int {
...
+	}
 
-			spawnable, flavor := s.checkQuota(reqForSpawn, requestedFlavor, call)
-			if spawnable == 0 {
-				return
-			}
+	// spawn servers in the background
+	s.Debug("spawnMultiple will spawn new servers", "cmd", cmd, "desired", desired, "perserver", perServer, "spawnable", spawnable, "allowed", allowed, "already", spawningCmd, "actual", todo)
+	for i := 0; i < todo; i++ {
+		s.spawningNow[cmd]++
```
Comment (kjsanger) [inline jobqueue/scheduler/openstack.go]: Is this access of `s.spawningNow[cmd]` outside a lock intentional?

File: cloud/server.go [hunk 2]
```diff
@@ -579,6 +629,41 @@ func (s *Server) CloseSSHSession(session *ssh.Session, clientIndex int) {
...
+	pw = &threadSafeWriteCLoser{WriteCloser: pw}
+	pr, err := s.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	return sftp.NewClientPipe(pr, pw)
+}
+
+type threadSafeWriteCLoser struct {
```
Comment (kjsanger) [inline cloud/server.go]: nit: typo `WriteCLoser`

File: cloud/cloud.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2016-2018 Genome Research Limited
+// Copyright © 2016-2020 Genome Research Limited
```
[the remark this answers is not in the document]
Comment (sb10) [reply cloud/cloud.go]: Actually, commits are in 2019 but not 2020. Corrected.

File: cloud/server.go (hunk 2 above)
Comment (sb10) [reply cloud/server.go]: Fixed

**Pull request merged**